### PR TITLE
Jv stream practice task solution

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -14,7 +18,14 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(s -> s.split(","))
+                .flatMap(Arrays::stream)
+                .mapToInt(Integer::parseInt)
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:24]: Try to use a minimum amount of stream operations
```suggestion
        return numbers.stream()
                .flatMap(...)
                .mapToInt(Integer::parseInt)
```

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -50,7 +67,13 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getAge() >= fromAge
+                        && ((person.getAge() <= maleToAge
+                        && person.getSex().equals(Person.Sex.MAN))
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:73]: image

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -75,6 +103,14 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(candidate -> new CandidateValidator().test(candidate))
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:107]:
```suggestion
                .filter(new CandidateValidator())
```

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -75,6 +103,14 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
...
+                .map(Candidate::getName)
+                .sorted()
+                .collect(Collectors.toList());
     }
 }
+
+
+
+
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:116]: remove empty lines

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request closed without merging**